src/test/java/module18/models/Pet.java

```diff
@@ -1,23 +1,24 @@
 package module18.models;
 
-import com.google.common.collect.ImmutableList;
+import module18.Statuses.PetStatus;
 
 import javax.swing.text.html.HTML;
 import java.util.List;
+import java.util.Objects;
 
 /**
  * Created by : Volodymyr_Silitskyi
  * Created at : 11/25/2018
  */
 
 
-public class Pet {
+public class Pet extends PetFactory {
```

Comment:
SOLID principles violation. 
Never do like this.  PetFactory should be separate class, Pet separate class. They do different things.
* PetFactory - class creator for Pet instances
* Pet - POJO class, describe Pet

And overall, inheritance in Java is an arguable principle and described by many people as OOD vulnerability #1.
I recommend to read Joshua Bloch `Effective Java 3rd Edition`:
* Item 18: Favor composition over inheritance,
* Item 19: Design and document for inheritance or else prohibit it